bfdd/bfd.c

```diff
@@ -677,6 +670,9 @@ struct bfd_session *ptm_bfd_sess_new(struct bfd_peer_cfg *bpc)
 	if (bpc->bpc_has_vrfname)
 		strlcpy(bfd->key.vrfname, bpc->bpc_vrfname,
 			sizeof(bfd->key.vrfname));
+	else
+		strlcpy(bfd->key.vrfname, VRF_DEFAULT_NAME,
+			sizeof(bfd->key.vrfname));
 
```

Comment:
same comment as previously. vrf default issue, when changing default vrf.

Reply:
actually, I think of some solutions:
- either rename the xpath forged with vrf=default to convert it to vrf=vrf0
(is it possible with nb api to do that?)
- or explicitly mention bfd session to configure with new vrf vrf0
- or use a "" value for default vrf name.

if the last solution could be possible, I think this could work; but with the yang definition, is it possible?

Reply:
any inputs for this, @rzalamena ?

Reply:
I have answered this before:

> We already were setting the default VRF implicitly, we were just not setting it in the key. Since the yang model requires VRF as the key for the peer I think it looks much better to have it point to the default VRF instead of empty string.

Also, I think this shouldn't be a problem. The default VRF is not show on the configuration (e.g. `show running` or `write`) and `VRF_DEFAULT_NAME` should adapt to this.

```c
#define VRF_DEFAULT_NAME vrf_get_default_name()
```

```
static char vrf_default_name[VRF_NAMSIZ] = VRF_DEFAULT_NAME_INTERNAL;
...
const char *vrf_get_default_name(void)
{
	return vrf_default_name;
}
```

Some questions to understand this better:

  1. Why would anyone change the default vrf name?
  2. Wouldn't this also break other daemons (have you tried `ripd`)?
  3. Shouldn't `default` be an alias to the default VRF name?

I think that maybe this is not a `bfdd` specific problem, but we can adapt it here if it is not too much work (this is a northbound migration PR after all...).